### ISSUE TITLE
Non-generic overload of SwaggerDocsConfig.MapType

### ISSUE THREAD

**Issue by jakubka** [domaindrivendev/Swashbuckle#643]
Hi,

Would it be possible to provide a non-generic overload of the MapType method?

Something like:

```
public void MapType(Type type, Func<Schema> factory)
{
    _customSchemaMappings.Add(type, factory);
}
```

It would help me to replace this nasty code with a loop:

```
config.MapType<Instant>(fun () -> schemas.Instant)
config.MapType<LocalDate>(fun () -> schemas.LocalDate)
config.MapType<LocalTime>(fun () -> schemas.LocalTime)
config.MapType<LocalDateTime>(fun () -> schemas.LocalDateTime)
config.MapType<OffsetDateTime>(fun () -> schemas.OffsetDateTime)
config.MapType<ZonedDateTime>(fun () -> schemas.ZonedDateTime)
config.MapType<Interval>(fun () -> schemas.Interval)
config.MapType<Offset>(fun () -> schemas.Offset)
config.MapType<Period>(fun () -> schemas.Period)
config.MapType<Duration>(fun () -> schemas.Duration)
config.MapType<DateTimeZone>(fun () -> schemas.DateTimeZone)
```

I can create PR if there is an interest.

Thanks!

**Comment by domaindrivendev** [comment]
Sounds reasonable - a PR would be awesome

**Comment by jakubka** [comment]
PR created: #653

**Comment by jakubka** [comment]
Hey, did you have a chance to check the PR? Thx

**Comment by domaindrivendev** [comment]
Merged and will be available when 5.4.0 get's released
